### PR TITLE
feat(ci): add PR path-based label routing and priority recommendation

### LINKED ISSUE #135: [Feature] Automate pull request routing and priority triage
## What task are you trying to do?

Keep the PawWork pull request review queue readable by automatically routing new PRs into the right type, scope, and suggested priority buckets. Maintainers should be able to see which PRs need immediate review, which are normal review work, and which can wait, without manually reclassifying every open PR after the queue grows.

## What do you do today?

Today PR priority is mostly manual. Recent open PRs skewed toward P2, which made the review queue hard to scan because low-risk docs, workflow, or UI enhancement PRs looked as urgent as product-path or release-path changes. Manual relabeling works for a small queue, but it does not scale when several feature and fix PRs are open at the same time.

## What would a good result look like?

New PRs should be routed automatically enough to reduce review pressure, while preserving human judgment for high-priority cases. A good first version would automatically apply type and scope labels from changed paths, add or comment a priority recommendation, default obvious low-risk PRs to P3, keep user-path or release-path changes at P2, and reserve P1/P0 for explicit maintainer confirmation. The routing should be explainable in the PR so reviewers can quickly see why a PR was classified that way.

## Which audience does this matter to most?

Not sure

## Extra context

Suggested implementation path: start with GitHub's official labeler action for type and scope labels, then add a conservative triage workflow using GitHub API logic or github-script. Initial rules can map .github/workflows/** to ci, packages/desktop-electron/** to desktop, packages/app/** to app, and UI-heavy app paths to ui. Priority should be conservative: docs, tests, small scripts, and isolated UI enhancements can be P3; app or desktop user-path changes can be P2; release blockers, CI blockers, security, runtime isolation, data loss, and updater or packaging failures should only become P1 after maintainer confirmation. CODEOWNERS can be considered later if reviewer assignment becomes necessary.

### PR DESCRIPTION
**Pull request by Astro-Han**
## Summary

Add two conservative PR triage automations:

- path-based scope routing via GitHub's official `actions/labeler`
- one-shot priority recommendation via `actions/github-script`, posting a review comment instead of auto-applying `P*` labels

This keeps maintainer control over `P0/P1` while reducing manual relabel work for routine PRs.

## Why

Issue #135 asks for a first-pass PR queue that is easier to scan without turning priority into an opaque bot decision. The safe v1 boundary is:

- automatically apply scope labels from changed paths
- recommend `P3` only for obvious low-risk PRs like docs, workflows, tests, and e2e-only changes
- recommend `P2` for app/desktop user-path changes and other non-low-risk paths
- never auto-apply a priority label, and never auto-suggest `P1/P0`

I also aligned the path routing with the **existing repo label taxonomy** instead of inventing new labels. Concretely:

- `.github/workflows/**` -> `ci`
- `packages/desktop-electron/**` -> `platform`
- `packages/app/**` -> `app`
- `packages/app/src/components/**`, `packages/ui/**`, `**/*.tsx` -> `ui`
- `packages/opencode/**` -> `harness`
- `docs/**`, `**/*.md` -> `documentation`

That preserves current label vocabulary while still matching the issue's routing intent.

## Related Issue

Closes #135.

## Human Review Status

Pending. A human should make the final merge decision after reviewing the final diff and verification evidence.

## Review Focus

- Whether the path-to-label mapping is the right conservative fit for the repo's **existing** labels.
- Whether the priority helper stays within the intended v1 boundary: only `P2/P3` recommendations, no auto-priority labels, no reviewer assignment.
- Whether the one-shot marker behavior is correct for `opened`, `synchronize`, and `reopened` without creating repeat triage noise.

## Risk Notes

- No product code, UI, or release workflow behavior changed.
- The new triage workflow runs on `pull_request_target`, but it checks out the PR base SHA only. It does not execute code from the PR head.
- Priority is posted as a review comment with a marker instead of mutating `P*` labels. That keeps maintainer override simple and avoids accidental `P1/P0` escalation.
- `actions/labeler` is configured against current repo labels (`platform`, `harness`, `documentation`) rather than the issue's placeholder names (`desktop`, `opencode`, `docs`) because those placeholder labels do not exist in the repo today.

## How To Verify

```text
actionlint .github/workflows/labeler.yml .github/workflows/pr-priority-triage.yml
Result: pass

bun --cwd packages/opencode test test/github/pr-routing-triage.test.ts
Result: 7 pass

bun --cwd packages/opencode typecheck
Result: pass

git diff --check
Result: pass

Sanity checks encoded in the contract test:
- PR #568 changed app src paths -> suggested P2
- PR #569 changed only e2e test paths -> suggested P3
- PR #571 changed only workflow + test paths -> suggested P3
```

## Screenshots or Recordings

Not needed. This is CI automation with no visible UI change.

## Checklist

- [x] Human review status is stated above as pending, approved, or not required
- [x] I linked the related issue, or stated why there is no issue
- [x] This PR has type, primary area, and priority labels, or I requested maintainer labeling
- [x] I described the review focus and any meaningful risks
- [x] I listed the relevant verification steps and the key result for each
- [x] I did not introduce unrelated refactors, dependencies, generated files, or file changes beyond the stated scope
- [x] I manually checked visible UI or copy changes when needed, with screenshots or recordings
- [x] I considered macOS and Windows impact for platform, packaging, updater, signing, paths, shell, or permissions changes
- [x] I called out docs, release notes, dependencies, permissions, credentials, deletion behavior, generated content, or local file changes when relevant
- [x] I reviewed the final diff for unrelated changes and suspicious dependency changes
- [x] I am targeting `dev`, and my PR title and commit messages use Conventional Commits in English

## Summary by CodeRabbit

* **Chores**
  * Added automated pull-request labeling rules covering CI, platform, app, UI, harness, and documentation.
  * Added workflows to sync labels and to post automated prioritization recommendations for PRs.
  * Added a helper that classifies changed files to suggest review priority.

* **Tests**
  * Added tests validating the label/triage workflows and the prioritization helper’s classifications and review content.

[Review Change Stack](https://app.coderabbit.ai/change-stack/Astro-Han/pawwork/pull/573)

### CODE REVIEW

**Review by Astro-Han — COMMENTED** [review]
## Opus second-pass review — PR #573

Pass, no blockers.

### Strengths

- Security pattern correct: `pull_request_target` + `actions/checkout` with `ref: github.event.pull_request.base.sha` ensures the workflow never runs PR-from-fork head code while still having access to the repo token. This is the recommended pattern for triage workflows.
- `persist-credentials: false` is fine here because both labeler and the github-script step use the API directly, not `git push` — no auth wiring needed.
- Concurrency group `pr-priority-triage-${pr.number}` with `cancel-in-progress: true` correctly handles rapid pushes, and the `TRIAGE_MARKER` dedup ensures duplicate comments cannot happen even with race-cancel.
- All third-party actions pinned to commit SHA (`actions/checkout@de0fac2e...`, `actions/github-script@f28e40c7...`, `actions/labeler@8558fd74...`), no `@v6` floating tags.
- Custom glob-to-regex in `pr-priority-triage.js` correctly handles `**/` prefix (`(?:.*/)?`) and bare `**` (`.*`). Traced `docs/**`, `**/*.md`, and `packages/app/src/**` — all behave as expected.
- Conservative v1 design: never auto-applies a P label, always comments a recommendation, P1/P0 reserved for maintainer. This is exactly the right trust model.
- Deliberate deviation from issue body: GPT chose real `platform/harness/documentation` labels over the issue body's placeholder `desktop/opencode/docs` because those don't exist in the repo. Documented in PR body. This is correct judgment — the workflow would have succeeded silently but applied no labels otherwise.
- `sync-labels: true` is safe in this setup because the labeler only manages the 6 path-based labels declared in `labeler.yml`; maintainer-added P labels and other labels (release-gate, windows, flaky-test, etc.) are untouched.
- Sanity cases in the contract test against recent real PRs are convincing:
  - PR #568 (production `packages/app/src/**` change) → P2 ✓
  - PR #569 (pure e2e test) → P3 ✓
  - PR #571 (CI workflow only) → P3 ✓
- 7 contract tests cover the classifier paths, marker dedup, and glob edge cases.

### Nit (no action)

- `pull_request_target` with `synchronize` re-runs on every push, but the marker dedup means subsequent pushes do NOT update the recommendation if the scope changes drastically (e.g., a PR that started doc-only and grew to product code). Acceptable for v1 — the comment is advisory and maintainers can manually relabel. If this becomes a pain point, a future iteration could compare current paths to the marker-encoded snapshot and re-post when scope drifts beyond a threshold.

- Could add an explicit note in PR body that `sync-labels: true` is intentionally limited to path-based labels and won't touch P-level or other manual labels. Helps future maintainers who scan the workflow and worry about label collisions.

### Verdict

CI workflow + path-based labeler + advisory comment surface. No product code, no release workflow change, no UI/UX. Conservative v1 design with sound security pattern and good test coverage. Ready for engineering final from @GPT-X.

File: .github/scripts/pr-priority-triage.js [hunk 1]
```diff
@@ -0,0 +1,95 @@
...
+      priority: "P2",
+      reason: `includes user-path files (${normalized.filter((path) => matchesAny(path, USER_PATH_GLOBS)).join(", ")})`,
+    }
+  }
+
+  return {
+    priority: "P2",
+    reason: "includes non-doc, non-test paths outside the low-risk bucket",
+  }
+}
```
[the remark this answers is not in the document]
Comment (Astro-Han) [reply .github/scripts/pr-priority-triage.js:81]: Thanks. I am deferring this in v1. The current goal is to keep the recommendation comment fully explainable from the exact changed paths, and the expected PR sizes here are still small enough that comment noise is an acceptable tradeoff for that transparency. If PR path lists start getting materially larger, truncating the display list is a reasonable follow-up refinement.

File: .github/scripts/pr-priority-triage.js [hunk 2]
```diff
@@ -0,0 +1,95 @@
...
+
+    out += escapeRegex(char)
+  }
+  out += "$"
+  return new RegExp(out)
+}
+
+function matchesAny(path, globs) {
+  return globs.some((glob) => globToRegExp(glob).test(path))
+}
```
[the remark this answers is not in the document]
Comment (Astro-Han) [reply .github/scripts/pr-priority-triage.js:51]: Thanks. I am not taking this in v1. This triage helper runs once per PR update and typical PawWork pull requests have a small enough path set that repeated regex compilation is not a practical cost. I want to keep the helper simple unless PR diff sizes grow enough to make this measurable. We can revisit memoization later if the workflow starts seeing larger path sets in practice.
